frontend/src/components/common/snackbar/SnackbarContext.tsx

```diff
@@ -0,0 +1,70 @@
+'use client';
+import type { ReactNode } from 'react';
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import Snackbar from '@mui/material/Snackbar';
+import Alert from '@mui/material/Alert';
+
+interface SnackbarContextType {
+  openSnackbar: (message: string, severity?: 'info' | 'warning') => void;
+  closeSnackbar: () => void;
+}
+
+const SnackbarContext = createContext<SnackbarContextType | undefined>(undefined);
+
+interface SnackbarProviderProps {
+  children: ReactNode;
+}
+
+export const SnackbarProvider: React.FC<SnackbarProviderProps> = ({ children }) => {
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState('');
+  const [snackbarSeverity, setSnackbarSeverity] = useState<'info' | 'warning' | undefined>(
+    undefined,
+  );
+
+  const openSnackbar = (message: string, severity?: 'info' | 'warning') => {
+    setSnackbarMessage(message);
+    setSnackbarSeverity(severity);
+    setSnackbarOpen(true);
+  };
+
+  const closeSnackbar = () => {
+    setSnackbarOpen(false);
+  };
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+    if (snackbarOpen) {
+      timeoutId = setTimeout(() => {
+        closeSnackbar();
+      }, 5000);
+    }
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [snackbarOpen]);
```

Comment:
This can be remove as it will close automatically with autoHideDuration
```
<Snackbar open={snackbarOpen} onClose={closeSnackbar} autoHideDuration={5000}>
```